src/views/bridge-details/bridge-details.view.tsx

```diff
@@ -38,23 +38,27 @@ interface Fees {
 
 const calculateFees = (bridge: Bridge): Promise<Fees> => {
   const step1Promise = bridge.from.provider
-    .getTransaction(
+    .getTransactionReceipt(
       bridge.status === "pending"
         ? bridge.claimTxHash || bridge.depositTxHash
         : bridge.depositTxHash
     )
-    .then((txResponse) => {
-      return txResponse ? calculateTransactionResponseFee(txResponse) : undefined;
-    })
+    .then((txReceipt) =>
+      txReceipt ? calculateTransactionReceiptFee({ txReceipt, type: "eip-1559" }) : undefined
+    )
     .catch(() => undefined);
 
   const step2Promise =
     bridge.status === "completed"
-      ? bridge.to.provider
-          .getTransaction(bridge.claimTxHash)
-          .then((txResponse) => {
-            return txResponse ? calculateTransactionResponseFee(txResponse) : undefined;
-          })
+      ? Promise.all([
+          bridge.to.provider.getTransactionReceipt(bridge.claimTxHash),
+          bridge.to.provider.getTransaction(bridge.claimTxHash),
+        ])
+          .then(([txReceipt, txResponse]) =>
+            txReceipt && txResponse
+              ? calculateTransactionReceiptFee({ txReceipt, txResponse, type: "legacy" })
+              : undefined
+          )
```

Comment:
Isn't this function assuming that step 1 (deposit) is L1>L2 (always using `eip-1559`) and step 2 (claim) is L2>L1 (always using `legacy`)?

I might be missing something but is this code also correct for a withdrawal? i.e step 1 occurs in L2 and step 2 occurs in L1?

Reply:
You are right! Would be great if I can differentiate an EIP-1559 tx receipt from a legacy one... Going to take a look to this

Reply:
Ah wait! I already know how to do it hehe